src/payout_queue/repo.rs

```diff
@@ -52,35 +52,31 @@ impl PayoutQueues {
         account_id: AccountId,
     ) -> Result<Vec<PayoutQueue>, PayoutQueueError> {
         let mut queues = Vec::new();
-        let mut query = Default::default();
+        let mut next = Some(PaginatedQueryArgs::default());
 
-        loop {
-            let mut paginated_queues = self
-                .list_for_account_id_by_id(account_id, query, Default::default())
+        while let Some(query) = next.take() {
+            let mut ret = self
```

Comment:
don't like `ret` here. That is short for `return`ed value... would prefer `res` - but its just a nit - can leave it like that.